corehq/apps/app_manager/static/app_manager/js/case_config_utils.js

```diff
@@ -48,24 +48,31 @@ hqDefine('app_manager/js/case_config_utils', function () {
             }
             return options;
         },
-        refreshQuestions: function (questions_observable, url, formUniqueId, event) {
-            var $el = $(event.currentTarget);
-            $el.find('i').addClass('fa-spin');
-            $.get({
-                url: url,
-                data: {
-                    form_unique_id: formUniqueId,
-                },
-                success: function (data) {
-                    questions_observable(data);
-                    $el.find('i').removeClass('fa-spin');
-                },
-                error: function () {
-                    $el.find('i').removeClass('fa-spin');
-                    hqImport("hqwebapp/js/alert_user").alert_user(gettext("Something went wrong refreshing "
-                               + "your form properties. Please refresh the page and try again", "danger"));
-                },
-            });
+        // This function depends on initial page data, so it should be called within a document ready handler
+        initRefreshQuestions: function (questionsObservable) {
+            var initialPageData = hqImport("hqwebapp/js/initial_page_data"),
+                formUniqueId = initialPageData.get("form_unique_id");
+            if (formUniqueId) {
+                var currentAppUrl = initialPageData.reverse("current_app_version"),
+                    oldVersion = initialPageData.get("app_subset").version;
+                $(document).on("ajaxComplete", function (e, xhr, options) {
+                    if (options.url === currentAppUrl) {
+                        var newVersion = xhr.responseJSON.currentVersion;
+                        if (newVersion > oldVersion) {
```

Comment:
Is it safer to check if `newVersion !== oldVersion`? Embarrassed to admit I don't know if it is possible to revert to an older version, but I imagine if an app was setup to use the latest _released_ version, it's pretty easy to revert to an older version.

Reply:
Totally reasonable question. When an app is reverted, it copies the older version into a new version. So even when reverting, the version number will never decrement. So `>` and `!==` should be equivalent here.